### PR TITLE
Add board support for Nologo ESP32S3 Pico

### PR DESCRIPTION
**Pull request by imyelo**
## Description of Change

Add board support for [Nologo ESP32S3 Pico](https://www.nologo.tech/product/esp32/esp32s3Pico/esp32S3Pico.html)

## Tests scenarios

I have tested my pr on this board. Verified by compiling and uploading using PlatformIO, cherry-picking the changes onto the v2.0.11 branch.

## Related links

- https://www.nologo.tech/product/esp32/esp32s3Pico/esp32S3Pico.html

### CODE REVIEW

File: boards.txt [hunk 1]
```diff
@@ -13818,6 +13818,247 @@ nodemcu-32s.menu.EraseFlash.all.upload.erase_cmd=-e
...
+nologo_esp32s3_pico.serial.disableDTR=false
+nologo_esp32s3_pico.serial.disableRTS=false
+
+nologo_esp32s3_pico.build.tarch=xtensa
+nologo_esp32s3_pico.build.bootloader_addr=0x0
+nologo_esp32s3_pico.build.target=esp32s3
+nologo_esp32s3_pico.build.mcu=esp32s3
+nologo_esp32s3_pico.build.core=esp32
+nologo_esp32s3_pico.build.variant=nologo_esp32s3_pico
+nologo_esp32s3_pico.build.board=nologo_esp32s3_pico
```
Comment (P-R-O-C-H-Y) [inline boards.txt:13848]: build.board must be uppercase.
```suggestion
nologo_esp32s3_pico.build.board=NOLOGO_ESP32S3_PICO
```

File: variants/nologo_esp32s3_pico/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#include <stdint.h>
+#include "soc/soc_caps.h"
+
+#define USB_VID 0x303a
+#define USB_PID 0x1001
+
+#define EXTERNAL_NUM_INTERRUPTS 28
+#define NUM_DIGITAL_PINS        48
+#define NUM_ANALOG_INPUTS       10
```
Comment (P-R-O-C-H-Y) [inline variants/nologo_esp32s3_pico/pins_arduino.h:12]: This section has been moved to `Arduino.h` file. Please remove it.

File: variants/nologo_esp32s3_pico/pins_arduino.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+static const uint8_t LED_BUILTIN = 21;
+#define BUILTIN_LED  LED_BUILTIN
+#define LED_BUILTIN LED_BUILTIN
+#define RGB_BUILTIN SOC_GPIO_PIN_COUNT + LED_BUILTIN
+#define RGB_BRIGHTNESS 64
+
+#define analogInputToDigitalPin(p)  (((p)<10)?(analogChannelToDigitalPin(p)):-1)
+#define digitalPinToInterrupt(p)    (((p)<49)?(p):-1)
+#define digitalPinHasPWM(p)         (p < 49)
```
Comment (P-R-O-C-H-Y) [inline variants/nologo_esp32s3_pico/pins_arduino.h:22]: This section has been moved to `Arduino.h` file. Please remove it.

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@imyelo PTAL on my comments.

**Review by P-R-O-C-H-Y — APPROVED** [review]

File: variants/nologo_esp32s3_pico/pins_arduino.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+static const uint8_t A1 = 2;
+static const uint8_t A2 = 3;
+static const uint8_t A3 = 4;
+static const uint8_t A4 = 5;
+static const uint8_t A5 = 6;
+static const uint8_t A6 = 7;
+static const uint8_t A7 = 8;
+static const uint8_t A8 = 9;
+static const uint8_t A9 = 10;
+
```
Comment (P-R-O-C-H-Y) [inline variants/nologo_esp32s3_pico/pins_arduino.h:26]: Compilation fails, as no I2C and SPI pins are declared. If you don't want to specify pins, you can declare them as -1.

```suggestion
// SPI - unused but you can create your own definition in your sketch
static const int8_t SCK = -1;
static const int8_t MISO = -1;
static const int8_t MOSI = -1;
static const int8_t SS = -1;

// I2C - unused but you can create your own definition in your sketch
static const uint8_t SDA = -1;
static const uint8_t SCL = -1;
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
PTAL @imyelo

File: variants/nologo_esp32s3_pico/pins_arduino.h (hunk 4 above)
Comment (imyelo) [reply variants/nologo_esp32s3_pico/pins_arduino.h:26]: Got it. Confirming with the manufacturer.

**Comment by P-R-O-C-H-Y** [comment]
@imyelo You have added a json file, which should not exist in our repository. Please remove it and add the I2C and SPI pins. Thanks

**Comment by imyelo** [comment]
> @imyelo You have added a json file, which should not exist in our repository. Please remove it and add the I2C and SPI pins. Thanks

Thanks for your review. Pins added, but I didn't find the json file mentioned. do you mean the reference in [the commit from another pr to platformio](platformio/platform-espressif32#1296), or I'm missing something :smile:

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Comment by P-R-O-C-H-Y** [comment]
Looks good now :) and I am 100% sure I saw the platformio json file in this PR before 😄.

**Comment by me-no-dev** [comment]
please fix the conflicts :)

**Comment by imyelo** [comment]
> please fix the conflicts :)

:ok_hand: Fixed.

**Pull request merged**